src/cpp/multisolver_interface/SolverXpress.cpp

```diff
@@ -37,26 +36,51 @@ SolverXpress::SolverXpress(const SolverLogManager& log_manager):
     }
 }
 
-SolverXpress::SolverXpress()
+namespace
 {
-    std::lock_guard<std::mutex> guard(license_guard);
-    int status = 0;
-    if (number_of_problems_counter() == 0)
+std::mutex& instance_guard()
+{
+    static std::mutex license_guard;
+    return license_guard;
+}
+} // namespace
+
+XpressManager::XpressManager()
+{
+    std::lock_guard<std::mutex> guard(instance_guard());
+    _loader.initXpressEnv();
+    if (_loader.XpressIsCorrectlyInstalled())
     {
-        LoadXpress::XpressLoader xpress_loader;
-        xpress_loader.initXpressEnv();
-        status = XPRSinit(NULL);
-        zero_status_check(status, "initialize XPRESS environment", LOGLOCATION);
+        int status = XPRSinit(nullptr);
+        SolverAbstract::zero_status_check(status, "initialize XPRESS environment", LOGLOCATION);
     }
+}
 
-    number_of_problems_counter() += 1;
+XpressManager::~XpressManager()
+{
+    std::lock_guard<std::mutex> guard(instance_guard());
```

Comment:
C++17 has [CTAD](https://en.cppreference.com/w/cpp/language/class_template_argument_deduction.html)
```suggestion
    std::lock_guard guard(instance_guard());
```